test/distributed/_tensor/test_utils.py

```diff
@@ -122,6 +123,17 @@ def test_compute_local_shape_and_global_offset_2D(self):
                     global_tensor[dim0_start:dim0_end, dim1_start:dim1_end],
                 )
 
+    @with_comms
+    def test_compute_global_padding(self):
+        local_tensor = torch.randn(7, 8, requires_grad=True)
+        mesh = init_device_mesh(self.device_type, (3, 2))
+        dist_tensor = distribute_tensor(local_tensor, mesh, [Shard(0), Shard(0)])
```

Comment:
in this test case, can you test various sharding combinations instead of only (Shard(0), Shard(0))? including the case where the num_shards exceed the tensor dim size